pandas/core/generic.py

```diff
@@ -10414,6 +10414,7 @@ def pct_change(self, periods=1, fill_method="pad", limit=None, freq=None, **kwar
             data = self.fillna(method=fill_method, limit=limit, axis=axis)
 
         rs = data.div(data.shift(periods=periods, freq=freq, axis=axis, **kwargs)) - 1
+        rs = rs.loc[~rs.index.duplicated()]
```

Comment:
I thought that handling the anchored freq was a key though, after the several testing, I have found that the approach I took was wrong.

so I take a new approach, dropping duplicate indices before reindexing like above.